back-end/services/loansService.js

```diff
@@ -1,15 +1,23 @@
-const { selectLoansByUserId } = require('./dbService')
+const { selectLoansByUserId, selectAllProductModels } = require('./dbService')
 
-const getLoansWithUserId = async userId => {
+const getLoansWithProductsByUserId = async userId => {
   try {
     const sqlLoans = await selectLoansByUserId(userId)
-    // waiting for Amélie`s function
-    // const sqlProducts =await selectProductModels()
+    const sqlProducts = await selectAllProductModels()
 
-    return sqlLoans
+    const result = sqlLoans.map(loan => {
+      return {
+        id: loan.id,
+        userId: loan.userId,
+        type: sqlProducts[loan.productId + 1],
```

Comment:
@amelieykw this doesn't look fool proof :D
or @JialinWang1 only `loan.productId` should be used it is now just a luck coincidence that the array index is almost the same as the id

Reply:
also I think it should be -1 to work at all